File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        i=0
        j=len(height)-1
        ma=0
        while i<j:
            h=min(height[i],height[j])
            a=h*(j-i)
            if a>ma:
                ma=a
            if height[i]<height[j]:
                i+=1
            else:
                j-=1
        return ma

File: 11-container-with-most-water/container-with-most-water.java
import java.util.*;
class Solution {
    public int maxArea(int[] height) {
        int i=0,j=height.length-1,ma=0;
        while(i<j){
            int h=Math.min(height[i],height[j]);
            int a=h*(j-i);
            if(a>ma){
                ma=a;
            }
            if (height[i]<height[j]){
                i++;
            }
            else{
                j--;
            }
        }
        return ma;
    }
}